src/app/error-boundary/ErrorBoundary.js

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router';
+import { Page404, Page500 } from '../../pages/errors';
+import styles from './ErrorBoundary.css';
+
+class ErrorBoundary extends Component {
+    static getDerivedStateFromProps(props, state) {
+        if (props.location.pathname !== state.pathname) {
+            return {
+                hasError: false,
+                error: undefined,
+                pathname: props.location.pathname,
+            };
+        }
+
+        return null;
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            hasError: false,
+            error: undefined,
+            pathname: props.location.pathname,
+        };
+    }
+
+    componentDidCatch(error) {
+        this.setState({ hasError: true, error });
+    }
+
+    render() {
+        const { hasError, error } = this.state;
+
+        if (hasError) {
+            switch (error.code) {
+            case 'UNKNOWN_IDENTITY':
+                document.body.classList.add(styles.hideErrorOverlay);
+
+                return <Page404 />;
+            default:
+                document.body.classList.remove(styles.hideErrorOverlay);
```

Comment:
Don't we need to also remove the class if we don't have an error?

Reply:
We don't need to remove the class because the iframe is not removed from the DOM. Removing the class would reveal the iframe again, and thus should only happen when we run into an unexpected error.